### PR TITLE
cnf-tests: support multiple hugepages size

### PR DESCRIPTION
**Pull request by SchSeba**
Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go [hunk 1]
```diff
@@ -845,27 +846,29 @@ func validatePerformanceProfile(performanceProfile *performancev2.PerformancePro
...
-	}
-
 	if len(performanceProfile.Spec.HugePages.Pages) == 0 {
 		return false, nil
 	}
 
-	if performanceProfile.Spec.HugePages.Pages[0].Count < 10 {
-		return false, nil
-	}
+	find1GHugePages := false
```
Comment (oribon) [inline cnf-tests/testsuites/e2esuite/dpdk/dpdk.go:853]: nit: found1GHugePages

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go [hunk 2]
```diff
@@ -845,27 +846,29 @@ func validatePerformanceProfile(performanceProfile *performancev2.PerformancePro
...
+	for idx := range performanceProfile.Spec.HugePages.Pages {
```
Comment (oribon) [inline cnf-tests/testsuites/e2esuite/dpdk/dpdk.go:854]: why use the index notation and not the item one here?

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go [hunk 3]
```diff
@@ -845,27 +846,29 @@ func validatePerformanceProfile(performanceProfile *performancev2.PerformancePro
...
+		countVerification := 5
+		// we need a minimum of 5 huge pages so if there is no Node we need 10 pages
```
Comment (oribon) [inline cnf-tests/testsuites/e2esuite/dpdk/dpdk.go:856]: can you explain a bit more please :sweat_smile: ?

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go [hunk 4]
```diff
@@ -845,27 +846,29 @@ func validatePerformanceProfile(performanceProfile *performancev2.PerformancePro
...
+		if performanceProfile.Spec.HugePages.Pages[idx].Size != "1G" {
+			continue
+		}
 
-	if performanceProfile.Spec.HugePages.Pages[0].Node != nil {
-		return false, nil
+		if performanceProfile.Spec.HugePages.Pages[idx].Count < int32(countVerification) {
+			continue
+		}
+		find1GHugePages = true
```
Comment (oribon) [inline cnf-tests/testsuites/e2esuite/dpdk/dpdk.go:868]: you can break here I think?

Comment (SchSeba) [reply cnf-tests/testsuites/e2esuite/dpdk/dpdk.go]: done

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go (hunk 3 above)
Comment (SchSeba) [reply cnf-tests/testsuites/e2esuite/dpdk/dpdk.go]: done

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go (hunk 2 above)
Comment (SchSeba) [reply cnf-tests/testsuites/e2esuite/dpdk/dpdk.go]: done

File: cnf-tests/testsuites/e2esuite/dpdk/dpdk.go (hunk 1 above)
Comment (SchSeba) [reply cnf-tests/testsuites/e2esuite/dpdk/dpdk.go]: done

**Pull request merged**